store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -91,6 +93,18 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
             throw VolumeServiceException.INVALID_REQUEST("Volume Compression Threshold can't be negative number", VolumeServiceException.BAD_VOLUME_COMPRESSION_THRESHOLD);
         }
 
+
+        if (Volume.TYPE_MESSAGE == volInfoRequest.getType() || Volume.TYPE_MESSAGE_SECONDARY == volInfoRequest.getType()) {
+            if (!StringUtil.isNullOrEmpty(volInfoRequest.getStoreManagerClass())){
+                if (!ClassHelper.isClassExist(volInfoRequest.getStoreManagerClass())) {
```

Comment:
can club both if into one

Reply:
Behavior will change then - no exception will be thrown if a non-empty class is specified but does not exist